tests/frontier/opcodes/test_swap.py

```diff
@@ -0,0 +1,130 @@
+    [getattr(Op, f"SWAP{i}") for i in range(1, 17)],  # Dynamically parametrize SWAP opcodes (SWAP1 to SWAP16)
+    ids=lambda op: str(op),
+)
+@pytest.mark.valid_from("Frontier")
+def test_swap(state_test: StateTestFiller, fork: Fork, pre: Alloc, swap_opcode: Op):
+    """
+    The set of `SWAP*` opcodes swaps the top of the stack with a specific element.
+
+    In this test, we ensure that the set of `SWAP*` opcodes correctly swaps
+    the top element with the nth element and stores the result in storage.
+    """
+    env = Environment()
+
+    # Calculate which position we're swapping with (1-based index)
+    swap_pos = swap_opcode.int() - 0x90 + 1
+
+    # Generate stack values
+    # We'll use position numbers as values to make it clear what's being swapped
+    # For SWAP1: [1, 0]
+    # For SWAP2: [2, 1, 0]
+    # etc.
+    stack_values = list(range(swap_pos + 1))
+
+    # Push the stack values onto the stack (in reverse order).
+    contract_code = Bytecode()
+    for value in reversed(stack_values):
+        contract_code += Op.PUSH1(value)
+
+    # Perform the SWAP operation.
+    contract_code += swap_opcode
+
+    # Store the top of the stack in storage slot 0.
+    contract_code += Op.PUSH1(0) + Op.SSTORE
+
+    # Deploy the contract with the generated bytecode.
+    contract = pre.deploy_contract(contract_code)
+
+    # Create a transaction to execute the contract.
+    tx = Transaction(
+        sender=pre.fund_eoa(),
+        to=contract,
+        gas_limit=500_000,
+        protected=False if fork in [Frontier, Homestead] else True,
+    )
+
+    # After SWAP, the top value will be the one at swap_pos
+    expected_value = swap_pos
+
+    # Define the expected post-state.
+    post = {}
+    post[contract] = Account(storage={0: expected_value})
+
+    # Run the state test.
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.parametrize(
+    "swap_opcode",
+    [getattr(Op, f"SWAP{i}") for i in range(1, 17)],
+    ids=lambda op: str(op),
+)
+@pytest.mark.parametrize("stack_height", range(1022, 1024))  # Test stack underflow conditions
+@pytest.mark.valid_from("Frontier")
+def test_stack_underflow(
+    state_test: StateTestFiller, fork: Fork, pre: Alloc, swap_opcode: Op, stack_height: int
+):
+    """
+    A test to ensure that the stack underflow when there are not enough elements
+    for the `SWAP*` opcode to operate.
+    """
+    env = Environment()
+
+    # Calculate which position we're swapping with (1-based index)
+    swap_pos = swap_opcode.int() - 0x90 + 1
+
+    # Push values onto the stack
+    contract_code = Bytecode()
+    # Push stack_height number of values, using the position as the value
+    for i in range(stack_height):
+        contract_code += Op.PUSH1(i % 256)  # Use modulo to keep values in byte range
+
+    # Attempt to perform the SWAP operation
+    contract_code += swap_opcode
+
+    # Store the top of the stack in storage slot 0
+    contract_code += Op.PUSH1(0) + Op.SSTORE
```

Comment:
We could change the approach a bit here:
- `SSTORE(0,1)` at the start of the execution, just to have something that will revert when the underflow happens.
- Fill the stack up to the `SWAP*` opcode's required minimum minus one.
- Perform the `SWAP*` that should abort the execution because there's not enough values in the stack for it to operate.
- Verify that the storage is empty since the `SSTORE(0,1)` reverts with the execution abort caused by the underflow.